src/Controller/Controller.php

```diff
@@ -124,6 +126,7 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      *
      * @var \Cake\Http\ServerRequest
      * @link https://book.cakephp.org/3.0/en/controllers/request-response.html#request
+     * @deprecated 3.5.0 Use getRequest()/setRequest instead.
```

Comment:
This is one of the places where I would strongly prefer `request()` for getting as that is the 99% case and everything else is annoying verbosity. See https://github.com/cakephp/cakephp/issues/10930.

I am not even sure if the request object should be changed? Should it? Or should it merely be cloned via `withFoo()`?

Reply:
@inoas You wouldn't need to use the methods *inside* the controller actions. In that context you could use the protected properties.